storage/block_storage.go

```diff
@@ -1078,3 +1077,54 @@ func (b *BlockStorage) AtTip(
 
 	return b.AtTipTransactional(ctx, tipDelay, transaction)
 }
+
+// IndexAtTip returns a boolean indicating if a block
+// index is at tip (provided some acceptable
+// tip delay). If the index is ahead of the head block
+// and the head block is at tip, we consider the
+// index at tip.
+func (b *BlockStorage) IndexAtTip(
+	ctx context.Context,
+	tipDelay int64,
+	index int64,
+) (bool, error) {
+	transaction := b.db.NewDatabaseTransaction(ctx, false)
+	defer transaction.Discard(ctx)
+	headBlockResponse, err := b.GetBlockLazyTransactional(ctx, nil, transaction)
+	if errors.Is(err, ErrHeadBlockNotFound) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+
+	// Check if index is greater than headBlock and if headBlock
+	// is at tip. If so, this is a result of us querying ahead of
+	// tip.
+	headBlock := headBlockResponse.Block
+	if headBlock.BlockIdentifier.Index < index {
+		if utils.AtTip(tipDelay, headBlock.Timestamp) {
```

Comment:
nit: can just do `return utils.AtTip(tipDelay, headBlock.Timstamp), nil`

Reply:
Nice!

Reply:
Usually my aggressive linter catches stuff like this.